### PR TITLE
Refactor Message - introduce enums

### PR DESCRIPTION
**Pull request by korotkevics**

### CODE REVIEW

File: shared/src/main/java/com/epam/ships/infra/communication/core/json/conversion/JSONDecoder.java [hunk 1]
```diff
@@ -1,37 +1,34 @@
...
      * @param jsonElement a JsonElement instance representing
-     *                   a BaseMessage.
+     *                   a Message.
      * @return Message a result of conversion of JsonElement
-     * into BaseMessage
+     * into Message
      */
     @Override
-    public Message decode(JsonElement jsonElement) {
+    public com.epam.ships.infra.communication.api.Message decode(JsonElement jsonElement) {
```
Comment (sabath91) [inline shared/src/main/java/com/epam/ships/infra/communication/core/json/conversion/JSONDecoder.java]: Remove fully qualified class name

File: shared/src/main/java/com/epam/ships/infra/communication/core/json/conversion/JSONEncoder.java [hunk 2]
```diff
@@ -30,8 +27,10 @@
      * implementing Message interface into a JsonElement instance.
      */
     @Override
-    public JsonElement encode(Message message) {
-        Gson gson = new GsonBuilder().enableComplexMapKeySerialization().create();
-        return gson.toJsonTree(message, BaseMessage.class);
+    public JsonElement encode(com.epam.ships.infra.communication.api.Message message) {
```
Comment (sabath91) [inline shared/src/main/java/com/epam/ships/infra/communication/core/json/conversion/JSONEncoder.java]: up

File: shared/src/main/java/com/epam/ships/infra/communication/core/message/Message.java [hunk 3]
```diff
@@ -23,18 +25,18 @@
 @EqualsAndHashCode
 @Getter(AccessLevel.PUBLIC)
 @Setter(AccessLevel.PACKAGE)
-public class BaseMessage implements Message {
+public class Message implements com.epam.ships.infra.communication.api.Message {
```
Comment (sabath91) [inline shared/src/main/java/com/epam/ships/infra/communication/core/message/Message.java]: as above

File: shared/src/main/java/com/epam/ships/infra/communication/core/message/MessageBuilder.java [hunk 4]
```diff
@@ -1,68 +1,70 @@
...
  */
 
 public class MessageBuilder {
 
-    private String header = StringUtils.EMPTY;
-    private String status = StringUtils.EMPTY;
-    private String author = StringUtils.EMPTY;
+    private com.epam.ships.infra.communication.api.message.Header header = Header.DEFAULT;
+    private com.epam.ships.infra.communication.api.message.Status status = Status.OK;
+    private com.epam.ships.infra.communication.api.message.Author author = Author.AUTO;
```
Comment (sabath91) [inline shared/src/main/java/com/epam/ships/infra/communication/core/message/MessageBuilder.java]: as above

File: shared/src/main/java/com/epam/ships/infra/communication/core/message/MessageBuilder.java [hunk 5]
```diff
@@ -1,68 +1,70 @@
...
      */
-    public final Message build() {
-        final BaseMessage baseMessage = new BaseMessage();
-        baseMessage.setHeader(this.header);
-        baseMessage.setStatus(this.status);
-        baseMessage.setAuthor(this.author);
-        baseMessage.setStatement(this.statement);
-        baseMessage.setFleet(this.fleet);
-        return baseMessage;
+    public final com.epam.ships.infra.communication.api.Message build() {
```
Comment (sabath91) [inline shared/src/main/java/com/epam/ships/infra/communication/core/message/MessageBuilder.java]: as above

File: shared/src/main/java/com/epam/ships/infra/communication/core/message/MessageBuilder.java [hunk 6]
```diff
@@ -1,68 +1,70 @@
...
+        return message;
     }
 
     /**
-     * @param header String value.
+     * @param header Header value.
      * @return a chain instance of MessageBuilder.
      */
-    public MessageBuilder withHeader(final String header) {
+    public MessageBuilder withHeader(final com.epam.ships.infra.communication.api.message.Header header) {
```
Comment (sabath91) [inline shared/src/main/java/com/epam/ships/infra/communication/core/message/MessageBuilder.java]: as above

**Comment by korotkevics** [comment]
Fixed

**Pull request closed without merging**